config/config.go

```diff
@@ -0,0 +1,73 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+type Config struct {
+	HomeDir string
+	BinDir  string
+	VarDir  string
+	RunDir  string
+
+	Telemetry struct {
+		Disable bool
+	}
+}
+
+type contextKey string
+
+var configContextKey = contextKey("config")
+
+func LoadDefault(cfg *Config) error {
+	if cfg.HomeDir == "" {
+		cfg.HomeDir = exoHome()
+	}
+
+	// TODO: make directory if necessary.
+	configFile := getExoPath(cfg.HomeDir, "config.toml", "EXO_CONFIG")
+	if err := loadFromFile(configFile, cfg); err != nil {
+		return err
+	}
+
+	if cfg.BinDir == "" {
+		cfg.BinDir = getExoPath(cfg.HomeDir, "bin", "EXO_BIN")
+	}
+	if cfg.RunDir == "" {
+		cfg.RunDir = getExoPath(cfg.HomeDir, "run", "EXO_RUN")
+	}
+	if cfg.VarDir == "" {
+		cfg.VarDir = getExoPath(cfg.HomeDir, "var", "EXO_VAR")
```

Comment:
Are we expecting to need these extra environment variables if/when we do packages for Homebrew, etc?

Reply:
No, I think that I can just use a flag or environment variable to specify an override location for the config file, then rely on that for any overrides.

Reply:
Yeah, I'd love to _minimize_ the customization and config that is possible :)